lib/core/providers/object/tools/text_tool_provider.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:ui';
+
+import 'package:paintroid/core/commands/graphic_factory/graphic_factory_provider.dart';
+import 'package:paintroid/core/enums/tool_types.dart';
+import 'package:paintroid/core/providers/state/canvas_state_provider.dart';
+import 'package:paintroid/core/tools/text_tool/bounding_box.dart';
+import 'package:paintroid/core/tools/text_tool/text_tool.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:paintroid/core/commands/command_factory/command_factory_provider.dart';
+import 'package:paintroid/core/commands/command_manager/command_manager_provider.dart';
+
+part 'text_tool_provider.g.dart';
+
+@riverpod
+class TextToolProvider extends _$TextToolProvider {
+  @override
+  TextTool build() {
+    Rect initialBoundingBox = Rect.fromCenter(
+      center: ref.read(canvasStateProvider).size.center(Offset.zero),
+      width: 300,
+      height: 200,
+    );
+    return TextTool(
+      graphicFactory: ref.watch(graphicFactoryProvider),
+      commandManager: ref.watch(commandManagerProvider),
+      commandFactory: ref.watch(commandFactoryProvider),
+      type: ToolType.TEXT,
+      boundingBox: BoundingBox(
+        initialBoundingBox.topLeft,
+        initialBoundingBox.topRight,
+        initialBoundingBox.bottomLeft,
+        initialBoundingBox.bottomRight,
+      ),
+    );
+  }
+
+  void updateText(String newText) {
+    state.currentText = newText;
+  }
```

Comment:
Please don't do side effects directly from this provider. Instead wherever you need it use something like this: 
```dart
final textTool = ref.read(toolBoxStateProvider).currentTool as TextTool;
textTool.currentText = newText
```

This was an example I think there should be a new provider to control the text of text tool to update the UI. So the `newText`can be retrieved from that new provider. Maybe a `ShapesToolOptionsStateProvider` ?